crates/federate/src/stats.rs

```diff
@@ -0,0 +1,65 @@
+use crate::util::get_latest_activity_id;
+use chrono::Local;
+use lemmy_api_common::federate_retry_sleep_duration;
+use lemmy_db_schema::{
+  source::federation_queue_state::FederationQueueState,
+  utils::{ActualDbPool, DbPool},
+};
+use std::{collections::HashMap, time::Duration};
+use tokio::{sync::mpsc::UnboundedReceiver, time::interval};
+use tracing::{debug, error, info};
+
+/// every 60s, print the state for every instance. exits if the receiver is done (all senders
+/// dropped)
+pub(crate) async fn receive_print_stats(
+  pool: ActualDbPool,
+  mut receiver: UnboundedReceiver<(String, FederationQueueState)>,
```

Comment:
Might as well make a struct for (domain, FederationQueueState), it makes it clear that the string is a domain.

Reply:
Ive changed it to use InstanceId instead of domain.

Reply:
not great because it now has an additional unnecessary db query every minute fetching 5k rows and an additional 25million rows it loops through every minute

Reply:
also doesn't make sense because the federationqueuestate already contains the instanceid, so now it's just passed in two times